### PR TITLE
[DOCS] Reformat get snapshot lifecycle policy API docs

### PR DESCRIPTION
**Pull request by jrodewig**
Updates the get snapshot lifecycle policy API docs to align with the new [API reference template](https://github.com/elastic/docs/blob/master/shared/api-ref-ex.asciidoc).

Relates to elastic/docs#937 and #45620

### CODE REVIEW

File: docs/reference/ilm/apis/slm-api.asciidoc [hunk 1]
```diff
@@ -110,28 +110,56 @@ To update an existing policy, simply use the put snapshot lifecycle policy API
...
+=== Get snapshot lifecycle policy API
+++++
+<titleabbrev>Get snapshot lifecycle policy</titleabbrev>
+++++
 
-Once a policy is in place, you can retrieve one or more of the policies using
-the get snapshot lifecycle policy API. This also includes information about the
-latest successful and failed invocation that the automatic snapshots have taken.
+Returns information
+about one or more snapshot lifecycle policies.
```
Comment (dakrone) [inline docs/reference/ilm/apis/slm-api.asciidoc]: What is up with the ~50 character line wrapping?

File: docs/reference/ilm/apis/slm-api.asciidoc [hunk 2]
```diff
@@ -169,17 +197,20 @@ The output looks similar to the following:
...
+[[slm-api-get-all-ex]]
+===== Get all policies
 
 [source,console]
 --------------------------------------------------
 GET /_slm/policy
 --------------------------------------------------
 // TEST[continued]
 
+
```
Comment (dakrone) [inline docs/reference/ilm/apis/slm-api.asciidoc]: Do we need double spacing between things for an asciidoctor reason?

**Review by dakrone — APPROVED** [review]
LGTM, though the more sterile tone does make me a bit sad

File: docs/reference/ilm/apis/slm-api.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/ilm/apis/slm-api.asciidoc]: I chiefly do this to better indicate breaks between sections. It has no impact on the final output.

**Comment by jrodewig** [comment]
Thanks for your review @dakrone.

If wanted, I can update this PR to keep most of the original wording and simply implement the new headings, etc.

I'm not trying to rob the docs of their character. 😄

File: docs/reference/ilm/apis/slm-api.asciidoc (hunk 1 above)
Comment (dakrone) [reply docs/reference/ilm/apis/slm-api.asciidoc]: I think it makes it much harder to review, but that may be just my personal preference.

**Comment by dakrone** [comment]
> If wanted, I can update this PR to keep most of the original wording and simply implement the new headings, etc.

That's up to the docs team, I understand if they want them in a certain voice. :)

File: docs/reference/ilm/apis/slm-api.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/ilm/apis/slm-api.asciidoc]: Thanks for the feedback!

**Pull request merged**